README.md
# nat.jrcbrains
<!-- badges: start -->
[![natverse](https://img.shields.io/badge/natverse-Part%20of%20the%20natverse-a241b6)](https://natverse.github.io)
[![Docs](https://img.shields.io/badge/docs-100%25-brightgreen.svg)](https://natverse.github.io/nat.jrcbrains/reference/)
[![Travis build status](https://travis-ci.org/natverse/nat.jrcbrains.svg?branch=master)](https://travis-ci.org/natverse/nat.jrcbrains)
<!-- badges: end -->

nat.jrcbrains allows bridging registrations for Drosophila brains from Janelia Research Campus
to be used with the [NeuroAnatomy Toolbox](https://natverse.github.io/nat/) suite
of R packages.



## Installation

You can install the development version of nat.jrcbrains from GitHub
using the remotes package (part of devtools):

``` r
if (!requireNamespace("remotes")) install.packages("remotes")
remotes::install_github('natverse/nat.jrcbrains')
```

## Example

You must first download the registrations you want from

* https://www.janelia.org/open-science/jrc-2018-brain-templates

This is most easily done as follows:

``` r
library(nat.jrcbrains)
download_saalfeldlab_registrations()
```

To run this example you will need *JRC 2018 Female - JFRC 2010*, which is
included as one of the downloaded  bridging registrations since 
[v0.3.1](https://github.com/natverse/nat.jrcbrains/commit/ebe1b464200af0440fb582f9fa26f12063629250)

``` r
library(nat.jrcbrains)
library(nat.templatebrains)
# sample neurons in JFRC2 space
pd2a = read.neurons("https://ars.els-cdn.com/content/image/1-s2.0-S0896627318307426-mmc5.zip")
# note use of via argument to insist on higher quality bridging registration
# to FAFB14
pd2a.fafb14=xform_brain(pd2a, sample=JFRC2, reference="FAFB14", via=c("FAFB14um"))
```
## Registration Formats
By default `nat.jrcbrains` will use registrations in the 
[Saalfeld h5 format](https://github.com/saalfeldlab/template-building/wiki/Hdf5-Deformation-fields).
As an end user, support for these registrations should be installed transparently
when you install `nat.jrcbrains`.

### ANTs format registrations
If you intend to use registrations in `ANTs` format, then you will need to install
the suggested [nat.ants](https://github.com/jefferis/nat.ants) package. 
This is turn has additional dependencies that can be hard to install. See the README for
https://github.com/jefferis/nat.ants for details but essentially doing

```
source("https://neuroconductor.org/neurocLite.R")
neuro_install("ANTsRCore")
remotes::install_github("jefferis/nat.ants")
```

should do the trick.

## Acknowledgements

This is a thin wrapper to allow convenient download and application of bridging
registrations developed by John Bogovic in [Stephan Saalfeld's lab](https://www.janelia.org/lab/saalfeld-lab) at Janelia.
See:

* https://www.biorxiv.org/content/10.1101/376384v2

*An unbiased template of the Drosophila brain and ventral nerve cord*. John A Bogovic, Hideo Otsuna, Larissa Heinrich, Masayoshi Ito, Jennifer Jeter, Geoffrey Meissner, Aljoscha Nern, Jennifer Colonell, Oz Malkesman, Kei Ito, Stephan Saalfeld
bioRxiv 376384; doi:[10.1101/376384](https://doi.org/10.1101/376384)

for details and additional downloads. **Please cite their paper if you make use of these registrations**.

In addition if you make use of the infrastructure in this and other natverse packages, we would be grateful if you would cite:

*The natverse, a versatile toolbox for combining and analysing neuroanatomical data.* A. S. Bates and J. D. Manton and S. R. Jagannathan and M. Costa and P. Schlegel and T. Rohlfing and G. S. X. E. Jefferis. eLife 9. doi:[10.7554/eLife.53350](https://doi.org/10.7554/eLife.53350)

You can also get citation information from inside R for the natverse
or individual packages like so:

```r
citation('natverse')
citation('nat.jrcbrains')
```

Your citations will help support future tool development by the Saalfeld lab and ourselves.


NEWS.md
# nat.jrcbrains 0.3.2

* add `mirror_fafb()` function (#11)

# nat.jrcbrains 0.3.1

* Add JRC2018F_JFRC2010 bridging registration. This enables bridging to the
  [JFRC2 template currently used by Virtual Fly Brain](https://github.com/VirtualFlyBrain/DrosAdultBRAINdomains). 
  However there are still some restrictions as noted in #10

# nat.jrcbrains 0.3.0

* Add JRC2018U bridging registration (#7)
* `download_saalfeldlab_registrations()`  now immediately activates downloaded
  registrations (#9)

# nat.jrcbrains 0.2.0

* support for downloading h5 or nifti format registrations.
* Added a `NEWS.md` file to track changes to the package

# nat.jrcbrains 0.1.0

* First skeleton version


tests/test-all.R
library(testthat)
library(nat.jrcbrains)

test_check("nat.jrcbrains")

# Run all test files
test_check("nat.jrcbrains")


tests/testthat/test-saalfeld_reg.R
context("Saalfeld_reg")
require(testthat)
library(nat)


########## Local functions for aiding test cases ####################

#local function for sampling points from a bounding box
sample_points_in_bb <- function(bb, n){
  mm=mapply(runif, min=bb[1,], max=bb[2,], n = n)
  colnames(mm)=c("X","Y","Z")
  mm
}

set.seed(42)
#Sample points to be used in test cases later
# dput(boundingbox(elmr::FAFB.surf))
fafb.bb = makeboundingbox(c(213175.6, 840529, 77531.3, 388177.9, 799.2, 269583.7))
test.pts.fafb <- sample_points_in_bb(fafb.bb,100)



########## h5 registration related test cases ####################
test_that("Check if h5 registration files can be downloaded", {

  skip("Skip it")
  skip_on_cran()
  skip_on_travis()
  skip_if_offline()


  path = getOption('nat.jrcbrains.regfolder')
  #Check if the .h5 files can be downloaded
  download_saalfeldlab_registrations()
  dir_list <- list.dirs(path = path, full.names = FALSE)
  dir_list <- dir_list[nchar(dir_list) != 0]
  for (download_fileidx in 1:length(dir_list)){
    file_name <- list.files(path = file.path(path,dir_list[download_fileidx]))
    expect_equal(file_name,paste0(dir_list[download_fileidx],'.h5'))
  }
})


test_that("Check if high resolution h5 registrations can be used", {
  skip_on_cran()
  skip_on_travis()
  status <- register_saalfeldlab_registrations()
  regnames <- basename(names(status))

  expect_equal(length(names(status)), length(status == TRUE)) #Check if all transforms were registered

  #JRC2018F_FAFB
  expect_silent(
    test.pts.jrc2018f <- nat.templatebrains::xform_brain(
      test.pts.fafb,
      sample = 'FAFB14',
      reference = nat.flybrains::JRC2018F))
  expect_silent(
    test.pts.fafb.t <- nat.templatebrains::xform_brain(test.pts.jrc2018f,
                                                       sample=nat.flybrains::JRC2018F,
                                                       reference='FAFB14',swap=TRUE))

  dists=sqrt(rowSums((as.matrix(test.pts.fafb)-as.matrix(test.pts.fafb.t))^2))
  expect_lt(mean(dists), 20) # 200 nm
})


context("Downsampled h5 tests")
test_that("Check if downsampled h5 registrations can be used", {

  testregfolder='testdata/downsampledreg'
  op <- options(nat.jrcbrains.regfolder=file.path(getwd(),test_path(),testregfolder))
  on.exit(options(op))

  message("Reg folder is: ", getOption('nat.jrcbrains.regfolder'))

  status <- register_saalfeldlab_registrations()
  #Check if all transforms were registered
  expect_true(all(status))

  #JRC2018F_FAFB
  expect_silent(
    test.pts.jrc2018f <- nat.templatebrains::xform_brain(
      test.pts.fafb,
      sample = 'FAFB14',
      reference = nat.flybrains::JRC2018F))
  expect_silent(
    test.pts.fafb.t <- nat.templatebrains::xform_brain(test.pts.jrc2018f,
                                                       sample=nat.flybrains::JRC2018F,
                                                       reference='FAFB14',swap=TRUE))

  dists=sqrt(rowSums((as.matrix(test.pts.fafb)-as.matrix(test.pts.fafb.t))^2))
  expect_lt(mean(dists), 43) # 200 nm
})

message("Reg folder is: ", getOption('nat.jrcbrains.regfolder'))

########## nii registration related test cases ####################

#Start afresh
# delete_saalfeldlab_registrations()

test_that("Check if nii registration files can be downloaded", {
  skip('Not now ...')
  skip_on_cran()
  skip_on_travis()
  skip_if_offline()

  path = getOption('nat.jrcbrains.regfolder')
  #Check if the .nii files can be downloaded
  download_saalfeldlab_registrations(fileformat = '.nii')
  dir_list <- list.dirs(path = path, full.names = FALSE)
  dir_list <- dir_list[nchar(dir_list) != 0]
  pattern <- "[_]*[0]*[_]*GenericAffine.mat$"
  for (download_fileidx in 1:length(dir_list)){
    affinefile <- list.files(path = file.path(path,dir_list[download_fileidx]), pattern = pattern)
    expect_false(isTRUE(all.equal(affinefile, character(0))))
  }

})


test_that("Check if the nii registrations can be used", {
  skip('Not now ...')
  status <- register_saalfeldlab_registrations()
  expect_equal(length(names(status)), length(status == TRUE)) #Check if all transforms were registered

  #JRC2018F_FAFB
  test.pts.jrc2018f<-nat.templatebrains::xform_brain(test.pts.fafb,
                                                                    sample=elmr::FAFB,
                                                                    reference=nat.flybrains::JRC2018F)
  test.pts.fafb.t <- nat.templatebrains::xform_brain(test.pts.jrc2018f,
                                                                    sample=nat.flybrains::JRC2018F,
                                                                    reference=elmr::FAFB,swap=TRUE)

  test.pts.fafb.t$inside <- NULL
  expect_equal(as.matrix(test.pts.fafb),as.matrix(test.pts.fafb.t),tolerance = 0.0001)

})


R/zzz.R

.onLoad <- function(libname, pkgname) {

  if (is.null(getOption('nat.jrcbrains.regfolder'))){
    options(nat.jrcbrains.regfolder=rappdirs::user_data_dir('R/nat.jrcbrains'))
  }

  suppressMessages(register_saalfeldlab_registrations())
  invisible()
}

.onAttach <- function(libname, pkgname) {
  # see if we've got anything that looks like our standard registrations
  havereg=FALSE
  regroot=getOption('nat.jrcbrains.regfolder')
  if(!is.null(regroot) && file.exists(regroot)) {
    # anything inside?
    havereg = length(dir(regroot))>0
  }

  if (interactive() && !havereg) {
    packageStartupMessage(
      "You do not have any Saalfeld lab registrations installed!\n",
      "We recommend getting them with:\n\n",
      "  download_saalfeldlab_registrations()\n\n",
      "If you think you already downloaded them, check the value of options('nat.jrcbrains.regfolder')\n",
      "which is currently set to: ", getOption('nat.jrcbrains.regfolder')
    )
  }
  invisible()
}


R/mirror.R
#' Mirror an object in FAFB space using FAFB-JRC2018F registration
#'
#' @param x An object containing 3D vertices (e.g. a neuron, surface, points)
#' @param sample The starting brain space in which \code{x} lives -
#'   \code{FAFB14} by default if it is not otherwise other specified.
#' @param via (optional, for expert use only) Intermediate template brain that
#'   the registration sequence must pass through. See details and \code{\link{xform_brain}}
#'   for more information.
#' @param ... Additional arguments passed to \code{\link{xform_brain}} and
#'   \code{\link{mirror_brain}}
#' @inheritParams nat.templatebrains::xform_brain
#' @inheritParams nat::nlapply
#'
#' @return A mirrored version of the neuron/surface or other 3D object.
#'
#' @details This function works by mapping the input data to \bold{JRC2018F}, a
#'   high quality symmetric female brain template. The data are then flipped
#'   (since \bold{JRC2018F} is symmetric no other action is required) and then
#'   returned to the starting space. In order to ensure that the Bogovic et al
#'   bridging registrations are preferred the via argument will be filled with
#'   the value "FAFB14um" i.e. \bold{FAFB14} calibrated in microns. This works
#'   because the Bogovic registrations were based on a mock synaptic staining
#'   using synapse locations that had been rescaled to microns before calculating
#'   a registration with ANTs to \bold{JRC2018F}.
#'
#' @importFrom nat.templatebrains xform_brain mirror_brain regtemplate
#' @export
#'
#' @examples
#' \donttest{
#' # transform arbitrary location (in nm)
#' mirror_fafb(cbind(388112, 162988, 132840))
#' # same but in units of microns
#' mirror_fafb(cbind(388.112, 162.988, 132.84), sample='FAFB14um')
#
#' # no messages
#' mirror_fafb(cbind(388112, 162988, 132840), Verbose=FALSE)
#'
#' # transform arbitrary location (from and to raw pixel coordinates)
#' voxel.size=c(4,4,40)
#' mirror_fafb(cbind(97028, 40747, 3321)* voxel.size)/c(voxel.size)
#'
#' library(nat.flybrains)
#' if(require('elmr', quietly = TRUE)) {
#' FAFB.surf.m <- mirror_fafb(FAFB14.surf)
#' wire3d(as.mesh3d(FAFB.surf), col='blue')
#' wire3d(as.mesh3d(FAFB.surf.m), col='red')
#' }
#'
#' \dontrun{
#' # you can also use with points in https://flywire.ai space
#' # converts from raw (voxel) coords used by Neuroglancer to nm and back again
#' voxel.size=c(4,4,40)
#' mirror_fafb(cbind(120508, 46292, 1667)*voxel.size, sample='FlyWire')/voxel.size
#' # compare with just using FAFB14
#' mirror_fafb(cbind(120508, 46292, 1667)*voxel.size)/voxel.size
#' }
#' }
mirror_fafb <- function(x, sample=NULL, subset=NULL, via=NULL, ...) {
  if(is.null(sample))
    sample <- regtemplate(x)
  if(is.null(sample))
    sample='FAFB14'
  if(is.null(via))
    via <- if(isTRUE(as.character(sample)=='FAFB14um')) NULL else 'FAFB14um'

  x.jrc2018 <- xform_brain(x, sample=sample, reference = nat.flybrains::JRC2018F,
                           subset=subset, via=via, ...)
  x.jrc2018.m <- mirror_brain(x.jrc2018, brain=nat.flybrains::JRC2018F,
                              transform = 'flip', subset=subset, ...)
  xt <- xform_brain(x.jrc2018.m, reference = sample, sample=nat.flybrains::JRC2018F,
                    subset=subset, via=via, ...)
  xt
}


R/saalfeld_reg.R
#' Download Saalfeld Lab registrations
#'
#' @param fileformat whether to download h5 (Saalfeld) or nii (ANTs) format
#'   registrations (defaults to h5).
#' @param activate whether to activate any new bridging registrations
#'   immediately for use by \code{\link{xform_brain}} and friends (default
#'   \code{TRUE}).
#' @return A character vector naming any new registrations downloaded,
#'   invisibly.
#' @details Registrations will be downloaded from
#'   \url{https://www.janelia.org/open-science/jrc-2018-brain-templates}. They
#'   will be downloaded to a folder defined by the package option
#'   \code{nat.templatebrains.regdirs} (see examples).
#'
#'   The h5 registration format typically provides a multi-resolution forward
#'   and inverse transformations in an efficiently compressed format. See
#'   \url{https://github.com/saalfeldlab/template-building/wiki/Hdf5-Deformation-fields}
#'    for further details.
#'
#'   The nii (NIfTI) files are the raw output format of the ANTs registration
#'   package. Their disadvantages are that they are 1) large files and the whole
#'   >500 Mb registration file must be loaded completely into memory before any
#'   transformations are possible, 2) that they depend on a more complex set of
#'   packages that can be hard to install, 3) is not simple to downsample to
#'   make smaller files. They do have the advantage of support for transforming
#'   image data as well as 3D points.
#'
#'
#' @export
#' @importFrom rappdirs user_data_dir
#' @importFrom curl curl_download
#' @seealso \code{\link{register_saalfeldlab_registrations}}
#' @examples
#' regroot=getOption('nat.templatebrains.regdirs')
#' dir(regroot)
download_saalfeldlab_registrations <- function(fileformat = c('.h5', '.nii'),
                                               activate=TRUE) {
  fileformat=match.arg(fileformat)

  if (fileformat == '.nii'){
    check_ants()
    #Support for JRC2018F_FAFB, JRC2018F_JFRC2013, JRC2018F_FCWB
    download_urls <-
      paste0(
        "https://ndownloader.figshare.com/files/",
        c(
          "12919949?private_link=85b2f2e4f479c94441f2",
          "12919832?private_link=a15a5cc56770ec340366",
          "12919868?private_link=6702242e17c564229874"
        )
      )

    download_filename <- rep('download_reg.zip', length(download_urls))
    search_pattern <- c("0GenericAffine.mat$","GenericAffine.mat$","0GenericAffine.mat$")
    regexpattern <- c("^([^_]+)_([^_]+)",
                      "^([^_]+)_([^_]+)_",
                      "^([^_]+)_([^_]+)_")
  } else if (fileformat == '.h5'){
    #Support for JRC2018F_FAFB, JRC2018F_JFRC2013, JRC2018F_FCWB, JRC2018U_JRC2018F
    download_urls <- paste0(

      "https://ndownloader.figshare.com/files/",
      c(
        "14362754?private_link=3a8b1d84c5e197edc97c",
        "14368703?private_link=2a684586d5014e31076c",
        "14369093?private_link=d5965dad295e46241ae1",
        "21749535?private_link=ca603876efb33fdf3028",
        "14371574?private_link=b7120207f38b35f1e372",
        "14368358?private_link=b29e25b6e47ccf9187a8"
      )
    )

    download_filename <-
      c(
        'JRC2018F_FAFB.h5',
        'JRC2018F_JFRC2013.h5',
        'JRC2018F_FCWB.h5',
        'JRC2018F_JRCFIB2018F.h5',
        'JRC2018U_JRC2018F.h5',
        'JRC2018F_JFRC2010.h5'
      )
    search_pattern <- rep(".h5$", length(download_urls))
    regexpattern <- rep("^([^_]+)_([^_]+)", length(download_urls))
  }

  #Step 1: check if folder path exists..
  if (!dir.exists(getOption('nat.jrcbrains.regfolder'))){
    message("Creating folder: ", getOption('nat.jrcbrains.regfolder'))
    dir.create(getOption('nat.jrcbrains.regfolder'), recursive = TRUE, showWarnings = FALSE)
  }

  updated=character()

  #Step 2: Download the files to the folder..
  if(length(download_urls))
    message("Downloading files to: ", getOption('nat.jrcbrains.regfolder'))
  for (download_fileidx in 1:length(download_urls)){
    message("Processing: ", download_filename[download_fileidx],
            " (file ", download_fileidx, "/", length(download_urls), ")")
    curl_download(download_urls[download_fileidx],
                  file.path(getOption('nat.jrcbrains.regfolder'),
                            download_filename[download_fileidx]),
                  quiet=FALSE)
    if (download_filename[download_fileidx] == 'download_reg.zip') {
      utils::unzip(file.path(getOption('nat.jrcbrains.regfolder'),
                             download_filename[download_fileidx]),
                   exdir = getOption('nat.jrcbrains.regfolder'))
      unlink(file.path(getOption('nat.jrcbrains.regfolder'),
                       download_filename[download_fileidx]))
    }

    file_name <- list.files(path = getOption('nat.jrcbrains.regfolder'),
                            pattern = search_pattern[download_fileidx], recursive = FALSE)
    matchnames <- stringr::str_match(file_name, paste0(regexpattern[download_fileidx],
                                                       search_pattern[download_fileidx]))
    folder_name <- paste0(matchnames[2],'_',matchnames[3])
    folder_path <- file.path(getOption('nat.jrcbrains.regfolder'),folder_name)
    dir.create(folder_path, recursive = FALSE, showWarnings = FALSE)

    files_target <- setdiff(list.files(getOption('nat.jrcbrains.regfolder'),folder_name),
                            folder_name)
    result <- file.copy(file.path(getOption('nat.jrcbrains.regfolder'),files_target),
              file.path(getOption('nat.jrcbrains.regfolder'),folder_name))
    if(any(result)){
      updated=union(updated, files_target)
    }
    unlink(file.path(getOption('nat.jrcbrains.regfolder'),files_target))
  }

  if(isTRUE(activate) && length(updated)>0){
    register_saalfeldlab_registrations()
  }
  invisible(updated)
}

#' Register Saalfeld Lab registrations with nat.templatebrains
#'
#' @details You must download the registrations from
#'   \url{https://www.janelia.org/open-science/jrc-2018-brain-templates} using
#'   \code{\link{download_saalfeldlab_registrations}}. If you prefer you may
#'   also download them manually to to a single folder, setting
#'   \code{options(nat.jrcbrains.regfolder)} to point to this folder in your
#'   \code{\link{Rprofile}}.
#' @param x A folder containing downloaded Saalfeld lab registrations
#' @param ... Additional arguments passed to
#'   \code{\link{add_saalfeldlab_reglist}} and eventually to
#'   \code{\link{add_reglist}}
#'
#' @return A named logical vector indicating which of the subfolders was
#'   successfully registered with \code{\link{nat.templatebrains}}
#' @seealso \code{\link{download_saalfeldlab_registrations}}
#' @export
#'
#' @examples
#' \dontrun{
#' options(nat.jrcbrains.regfolder='/GD/projects/JFRC/JohnBogovic/jrc-2018-brain-templates')
#' register_saalfeldlab_registrations()
#' }
#' @references See Bogovic et al. (2018) \url{https://doi.org/10.1101/376384}
register_saalfeldlab_registrations <- function(x=getOption('nat.jrcbrains.regfolder'),
                                               ...) {
  if(is.null(x)) stop("You must pass a folder containing registrations or set\n",
                      "options(nat.jrcbrains.regfolder='/path/to/reg/folder')")

  dirs=list_saalfeldlab_registrations(x)
  sapply(dirs, add_saalfeldlab_reglist, ...)
}

list_saalfeldlab_registrations <- function(x=getOption('nat.jrcbrains.regfolder')) {
  list.dirs(x, recursive = F)
}

#' Register single ANTs registration folder
#'
#' @param x Path to a single ANTs registration folder named according to
#'   Saalfeld conventions
#' @param ... Additional arguments passed to \code{\link{add_reglist}}
#'
#' @importFrom nat reglist
#' @importFrom nat.templatebrains add_reglist
add_saalfeldlab_reglist <- function(x, ...) {
  reg=foldertoreg(x)
  bx = basename(x)
  status <-  FALSE
  brainnames = stringr::str_match(bx, "^([^_]+)_([^_]+)$")
  if (any(is.na(brainnames))) {
    warning("Unable to identify the brain spaces linked by: ", bx)
    return(status)
  }

  # These registrations all assume that FAFB is calibrate in microns,
  # whereas FAFB14 registration
  if (any(brainnames == 'FAFB')) {
    brainnames[brainnames == "FAFB"] = 'FAFB14um'
    add_reglist(reglist(diag(c(1e3, 1e3, 1e3, 1))),
                reference = "FAFB14",
                sample = "FAFB14um")
  }

  # Bogovic et al seem to have a difference in Z calibration
  if (any(brainnames == 'JFRC2010')) {
    add_reglist(reglist(diag(c(1, 1, 1/0.6220880, 1))),
                reference = "JFRC2010",
                sample = "JFRC2")
  }

  add_reglist(reglist(reg),
              reference = brainnames[2],
              sample = brainnames[3], ...)
  message("Adding ",class(reg) ," in " , "forward direction")
  reg2=foldertoreg(x, inverse = TRUE)

  add_reglist(reglist(reg2),
              reference = brainnames[3],
              sample = brainnames[2],
              ...)
  message("Adding ",class(reg2) ," in " , "reverse direction")
  status <-  TRUE
}
#

foldertoreg <- function(folder, inverse=FALSE){
  ff = dir(folder, full.names = TRUE)
  exts= tools::file_ext(ff)
  if(any('h5' %in% exts)){
    h5file=ff[exts=='h5']

    if(length(h5file) >1)
      message("More than one .h5 file is present in ", folder)
    reg=nat.h5reg::h5reg(h5file, swap=inverse)
  } else {
    check_ants()
    reg = try(nat.ants::as.antsreg(folder, inverse=inverse), silent = TRUE)
    if (inherits(reg, 'try-error')) {
      stop(folder, " does not seem to be a valid",
           ifelse(inverse, "inverse", "forward"),
           "ANTs registration folder!")
    }
  }
  reg
}

check_ants <- function() {
  if(!requireNamespace('nat.ants', quietly = TRUE))
    stop("You must install the nat.ants package in order to use ANTs (nii) format registrations!\n",
         "Please see https://github.com/jefferis/nat.ants for details")
}

delete_saalfeldlab_registrations <- function(){
  x=getOption('nat.jrcbrains.regfolder')
  unlink(file.path(x,'.'),recursive = TRUE)
}
